internal/testhelpers/e2e_server.go

```diff
@@ -181,35 +181,44 @@ func CheckE2EServerOnHTTPS(t *testing.T, publicPort, adminPort int) (publicUrl,
 	return
 }
 
-func GenerateTLSCertificateFilesForTests(t *testing.T, certPath, keyPath string) {
+// GenerateTLSCertificateFilesForTests writes a new, self-signed TLS
+// certificate+key (in PEM format) to a temporary location on disk and returns
+// the paths to both. The files are automatically cleaned up when the given
+// *testing.T concludes its tests.
+func GenerateTLSCertificateFilesForTests(t *testing.T) (certPath, keyPath string) {
```

Comment:
As we also often need the base64 encoding of the PEM content, why not return it here as well and avoid repeatedly encoding it?

Reply:
streamlined the helper